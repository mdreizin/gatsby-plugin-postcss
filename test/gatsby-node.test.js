describe('onCreateWebpackConfig', () => {
  const actions = {
    setWebpackConfig: jest.fn(),
    replaceWebpackConfig: jest.fn()
  };
  const loaders = {
    miniCssExtract: () => `miniCssExtract`,
    null: () => 'null-loader',
    css: args => `css-loader(${JSON.stringify(args)})`
  };
  const getConfig = jest.fn().mockReturnValue({
    module: {
      rules: [
        {
          oneOf: ['postcss-loader']
        },
        {
          test: /\.js/,
          use: ['babel-loader']
        }
      ]
    }
  });
  const { onCreateWebpackConfig } = require('../src/gatsby-node');

  beforeEach(() => {
    actions.setWebpackConfig.mockReset();
  });

  const tests = {
    stages: ['develop', 'build-javascript', 'develop-html', 'build-html'],
    options: {
      'No options': {},
      'PostCss options': {
        postcss: { plugins: ['autoprefixer'], sourceMap: false }
      }
    }
  };

  tests.stages.forEach(stage => {
    for (const label in tests.options) {
      const options = tests.options[label];

      it(`Stage: ${stage} / ${label}`, () => {
        onCreateWebpackConfig({ actions, loaders, stage, getConfig }, options);

        expect(actions.setWebpackConfig).toMatchSnapshot();
      });
    }
  });
});
