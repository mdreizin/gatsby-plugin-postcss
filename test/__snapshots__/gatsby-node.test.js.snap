// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`onCreateWebpackConfig Stage: build-html / No options 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "module": Object {
          "rules": Array [
            Object {
              "oneOf": Array [
                Object {
                  "test": /\\\\\\.module\\\\\\.css\\$/,
                  "use": Array [
                    "miniCssExtract",
                    "css-loader({\\"modules\\":true,\\"importLoaders\\":1})",
                    Object {
                      "loader": "postcss-loader",
                      "options": Object {
                        "sourceMap": false,
                      },
                    },
                  ],
                },
                Object {
                  "test": /\\\\\\.css\\$/,
                  "use": Array [
                    "null-loader",
                  ],
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;

exports[`onCreateWebpackConfig Stage: build-html / PostCss options 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "module": Object {
          "rules": Array [
            Object {
              "oneOf": Array [
                Object {
                  "test": /\\\\\\.module\\\\\\.css\\$/,
                  "use": Array [
                    "miniCssExtract",
                    "css-loader({\\"modules\\":true,\\"importLoaders\\":1})",
                    Object {
                      "loader": "postcss-loader",
                      "options": Object {
                        "plugins": Array [
                          "autoprefixer",
                        ],
                        "sourceMap": false,
                      },
                    },
                  ],
                },
                Object {
                  "test": /\\\\\\.css\\$/,
                  "use": Array [
                    "null-loader",
                  ],
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;

exports[`onCreateWebpackConfig Stage: build-javascript / No options 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "module": Object {
          "rules": Array [
            Object {
              "oneOf": Array [
                Object {
                  "test": /\\\\\\.module\\\\\\.css\\$/,
                  "use": Array [
                    "miniCssExtract",
                    "css-loader({\\"modules\\":true,\\"importLoaders\\":1})",
                    Object {
                      "loader": "postcss-loader",
                      "options": Object {
                        "sourceMap": false,
                      },
                    },
                  ],
                },
                Object {
                  "test": /\\\\\\.css\\$/,
                  "use": Array [
                    "miniCssExtract",
                    "css-loader({\\"importLoaders\\":1})",
                    Object {
                      "loader": "postcss-loader",
                      "options": Object {
                        "sourceMap": false,
                      },
                    },
                  ],
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;

exports[`onCreateWebpackConfig Stage: build-javascript / PostCss options 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "module": Object {
          "rules": Array [
            Object {
              "oneOf": Array [
                Object {
                  "test": /\\\\\\.module\\\\\\.css\\$/,
                  "use": Array [
                    "miniCssExtract",
                    "css-loader({\\"modules\\":true,\\"importLoaders\\":1})",
                    Object {
                      "loader": "postcss-loader",
                      "options": Object {
                        "plugins": Array [
                          "autoprefixer",
                        ],
                        "sourceMap": false,
                      },
                    },
                  ],
                },
                Object {
                  "test": /\\\\\\.css\\$/,
                  "use": Array [
                    "miniCssExtract",
                    "css-loader({\\"importLoaders\\":1})",
                    Object {
                      "loader": "postcss-loader",
                      "options": Object {
                        "plugins": Array [
                          "autoprefixer",
                        ],
                        "sourceMap": false,
                      },
                    },
                  ],
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;

exports[`onCreateWebpackConfig Stage: develop / No options 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "module": Object {
          "rules": Array [
            Object {
              "oneOf": Array [
                Object {
                  "test": /\\\\\\.module\\\\\\.css\\$/,
                  "use": Array [
                    "miniCssExtract",
                    "css-loader({\\"modules\\":true,\\"importLoaders\\":1})",
                    Object {
                      "loader": "postcss-loader",
                      "options": Object {
                        "sourceMap": true,
                      },
                    },
                  ],
                },
                Object {
                  "test": /\\\\\\.css\\$/,
                  "use": Array [
                    "miniCssExtract",
                    "css-loader({\\"importLoaders\\":1})",
                    Object {
                      "loader": "postcss-loader",
                      "options": Object {
                        "sourceMap": true,
                      },
                    },
                  ],
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;

exports[`onCreateWebpackConfig Stage: develop / PostCss options 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "module": Object {
          "rules": Array [
            Object {
              "oneOf": Array [
                Object {
                  "test": /\\\\\\.module\\\\\\.css\\$/,
                  "use": Array [
                    "miniCssExtract",
                    "css-loader({\\"modules\\":true,\\"importLoaders\\":1})",
                    Object {
                      "loader": "postcss-loader",
                      "options": Object {
                        "plugins": Array [
                          "autoprefixer",
                        ],
                        "sourceMap": false,
                      },
                    },
                  ],
                },
                Object {
                  "test": /\\\\\\.css\\$/,
                  "use": Array [
                    "miniCssExtract",
                    "css-loader({\\"importLoaders\\":1})",
                    Object {
                      "loader": "postcss-loader",
                      "options": Object {
                        "plugins": Array [
                          "autoprefixer",
                        ],
                        "sourceMap": false,
                      },
                    },
                  ],
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;

exports[`onCreateWebpackConfig Stage: develop-html / No options 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "module": Object {
          "rules": Array [
            Object {
              "oneOf": Array [
                Object {
                  "test": /\\\\\\.module\\\\\\.css\\$/,
                  "use": Array [
                    "miniCssExtract",
                    "css-loader({\\"modules\\":true,\\"importLoaders\\":1})",
                    Object {
                      "loader": "postcss-loader",
                      "options": Object {
                        "sourceMap": false,
                      },
                    },
                  ],
                },
                Object {
                  "test": /\\\\\\.css\\$/,
                  "use": Array [
                    "null-loader",
                  ],
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;

exports[`onCreateWebpackConfig Stage: develop-html / PostCss options 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "module": Object {
          "rules": Array [
            Object {
              "oneOf": Array [
                Object {
                  "test": /\\\\\\.module\\\\\\.css\\$/,
                  "use": Array [
                    "miniCssExtract",
                    "css-loader({\\"modules\\":true,\\"importLoaders\\":1})",
                    Object {
                      "loader": "postcss-loader",
                      "options": Object {
                        "plugins": Array [
                          "autoprefixer",
                        ],
                        "sourceMap": false,
                      },
                    },
                  ],
                },
                Object {
                  "test": /\\\\\\.css\\$/,
                  "use": Array [
                    "null-loader",
                  ],
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;
